plugins/more_selectors/more_selectors/main.py

```diff
@@ -0,0 +1,177 @@
+import cadquery as cq
+from math import sqrt
+from . import utils
+
+
+class InfiniteCylinderSelector(cq.Selector):
+    """
+    Selects any shape present in the defined infinite cylinder 
+    based on the shape center of mass point.   
+
+    """
+    def __init__(self, origin, along_axis, radius, debug=False):
+        self.outer_radius = radius
+        self.axis = self.get_axis(along_axis)
+        xdir = self.get_ortho_vector(self.axis)
+        self.base = cq.Plane(cq.Vector(origin), xdir, self.axis.toTuple())
+        if debug:
+            utils.make_debug_cylinder(self.base, self.outer_radius)
+
+    def get_axis(self, axis_value):
+        named_vectors = {
+            "X" : cq.Vector(1,0,0),
+            "-X" : cq.Vector(-1,0,0),
+            "Y" : cq.Vector(0,1,0),
+            "-Y" : cq.Vector(0,-1,0),
+            "Z" : cq.Vector(0,0,1),
+            "-Z" : cq.Vector(0,0,-1)
+        }
+        if isinstance(axis_value, str):
+            try:
+                return named_vectors[axis_value]
+            except KeyError:
+                raise ValueError("Supported names are {}".format(list(named_vectors.keys())))    
+        else:
+            if cq.Vector(axis_value).Length != 0:
+                return cq.Vector(axis_value)
+            else:
+                raise ValueError("Axis of the cylinder must be non null")
+
+    def get_ortho_vector(self, vector):
+        #returns a vector orthogonal to the provided vector
+        if 0 not in vector.toTuple():
+            return (1,1, (-vector.x - vector.y)/vector.z)    
+        else:
+            if vector.x != 0:
+                return ((-vector.y - vector.z)/vector.x, 1, 1)
+            elif vector.y != 0:
+                return (1, (-vector.x - vector.z)/vector.y,1)
+            else:
+                return (1,1, (-vector.x - vector.y)/vector.z)
```

Comment:
Are you aware of the `cadquery.Vector.cross` method? 
```suggestion
        other = cq.Vector(0, 1, 0)
        if vector.normalized() in [other, -other]:
            other = cq.Vector(1, 0, 0)
        return vector.cross(other)
```